### ISSUE TITLE
No option to tweak minimap textures server-side

### ISSUE THREAD

**Issue by BetterToAutomateTheWorld** [minetest/minetest#3098]
This is a feature request.

Like a texture pack, i think we need a server side option to add the possibility to tweak the texture of the new minimap, until the players doesn't add his own texture to force his texture instead of the server side texture (like texture pack)

Linked to one of our issue : https://github.com/MinetestForFun/minetest-minetestforfun-server/issues/230

Thank you in advance.

**Comment by ExeterDad** [comment]
I'd like to see this added as well.

**Comment by rubenwardy** [comment]
You can have server side texture packs, although the client can override it.

**Comment by Ombridride** [comment]
I don't understand your answer @rubenwardy 

This issue isn't about texture packs, it's about minimap override textures (but the client will always have the priority for his texture, that's normal)

We have tried to add the texture to the server, but it doesn't work the same way as textures pack. The texture isn't displayed at all.

**Comment by ExeterDad** [comment]
I've been unable to serve the texture in the way you describe @rubenwardy, Thus my interest in this issue.

**Comment by rubenwardy** [comment]
does it work if you supply eg default_dirt.png in a texture pack on the server?

**Comment by rubenwardy** [comment]
Confirmed. You can override default_dirt in `textures/server` but not the minimap overlay. 
I consider this a bug, not a feature request.

Here is where the texture is loaded: https://github.com/minetest/minetest/blob/master/src/minimap.cpp#L241

**Comment by JamesTobin** [comment]
@rubenwardy is this fixed in 5.1 perhaps? right now in my server, which disables minimap via a mod, 5.1 clients have no minimap, but 5.0 clients do. This is quite problematic, and would be good if you can fix this right away for 5.0 as well, as my 5.0 users are cheating the game

**Comment by Wuzzy2** [comment]
Do I understand this right? Is this issue about mods and games being unable to specify their own minimap textures like `object_marker.png`, `minimap_mask_round.png`, etc? Because that's the case. These files are only recognized from the texture pack but not from the game.

I tried to figure out why this is the case, but there's nothing suspect I can find in the code ... :-/

**Comment by sfan5** [comment]
IIRC the minimap code directly tries to load textures from the driver instead of going through our TextureSource which works (sort of) but is wrong.
